core/jobs/batch_jobs/cron_jobs.py

```diff
@@ -307,10 +302,10 @@ def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
             | 'Generate stats' >> beam.ParDo(
                 lambda x: self._generate_stats(
                     x['suggestion'][0] if len(x['suggestion']) else [],
-                    x['opportunity'][0][0] if len(x['opportunity']) else None
+                    list(x['opportunity'][0])[0]
```

Comment:
Looks like this was a legit error. One question -- how hard was it to debug this? Were the logs easily available and was it easy to connect the logs to the problem?

@brianrodri If we don't already, I think we should have a section on the wiki re "how to debug errors in Beam jobs", with a worked example. This seems like a good one.

Also how did this pass locally?

Reply:
This was passing locally since I resolved it a bit different way. It was quite hard to debug actually, I think this should be documented.